File: src/main/java/net/felsstudio/fels/parser/Token.java
package main.java.net.felsstudio.fels.parser;

public record Token(TokenType type, String text, Pos pos) {

    public String shortDescription() {
        return type().name() + " " + text;
    }

    @Override
    public String toString() {
        return type.name() + " " + pos().format() + " " + text;
    }
}

File: src/main/java/net/felsstudio/fels/parser/Token.kt
package main.java.net.felsstudio.fels.parser

/**
 *
 * @author felek
 */
class Token(@JvmField val type: TokenType, @JvmField val text: String, @JvmField val row: Int, val col: Int) {
    fun position(): String {
        return "[$row $col]"
    }

    override fun toString(): String {
        return type.name + " " + position() + " " + text
    }
}
